libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/util/RouteLineUtils.kt

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.navigation.ui.maps.util
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+
+/**
+ * Extension functions and utilities for route line related operations.
+ */
+object RouteLineUtils {
+
+    /**
+     * An extension function to reorder a collection of [DirectionsRoute] objects so that the
+     * individual route provided is the first in the collection indicating it is the primary route.
+     *
+     * This function can be useful when the user is selecting alternate routes. The order of
+     * operations would be to:
+     *
+     * Find the route selected via [MapboxRouteLineApi.findClosestRoute].
+     * Use this function to reorder the routes from [MapboxRouteLineApi.getRoutes].
+     * Then call [MapboxNavigation.setRoutes] to update the navigation core.
+     *
+     * Note: In many cases a [RoutesObserver] will be utilized in an activity or fragment which
+     * will include a call to [MapboxcRouteLineApi.setRoutes]. This call will update the state
+     * of the [MapboxcRouteLineApi] and rendering the result will update the route lines displayed
+     * on the map.
+     *
+     * If your application is not using a [RoutesObserver] to set the routes in [MapboxcRouteLineApi]
+     * then you should call [MapboxcRouteLineApi.setRoutes] in your activity or fragment with the
+     * collection returned by this function.
+     *
+     * @param route the route that should appear first in the collection indicating it is the
+     * primary route. If the route is not already in the collection it will be added as the first
+     * route in the collection.
+     */
+    @JvmStatic
+    fun List<DirectionsRoute>.updatePrimaryRoute(route: DirectionsRoute): List<DirectionsRoute> {
+        return this.filter { it != route }.toMutableList().also {
+            it.add(0, route)
+        }
```

Comment:
Let's check if the provided list of routes contained the new `route` in the first place. If not, we should probably return the original list and log an error.

Alternatively, we can return an `Expected<String, List<DirectionsRoute>>`.

Reply:
Why would it be necessary to check? I couldn't think of a reason it would be a problem if the inputted route wasn't already present. No external state is mutated and the resulting list of routes has the inputted route as the first in the collection which would be the intended result regardless of whether or not the inputted route was in the collection or not.

Reply:
The SDK supports a list of up to 3 routes, or at least the RouteLineApi does. If the list suddenly grows to 4 elements that could cause various issues, couldn't it?

Also we say ourselves in the docs
> An extension function to reorder a collection of [DirectionsRoute] objects

I just think that if someone uses a method that swaps the order in a list (which is what selecting an alternative practically means), they wouldn't want to add new elements to that list. If they want to re-arrange a list using an element that's not there, that sounds like an error to me, and allowing for that behavior could just obfuscate bugs in the future.

Reply:
The docs also say: If the route is not already in the collection it will be added as the first route in the collection.

The route line API will display a primary route and up to two alternative routes. Passing a list of 10 routes won't break anything. The additional routes just won't be displayed. Calling this extension function with a route that is or is not already in the collection has the same result (in the context of the intended usage), the passed in route is first in the collection. That doesn't seem erroneous to me. 

Changing the function so that it returns Expected<String, List<DirectionsRoute>> when the inputted route isn't in the collection seems like an overly complex way of doing something really simple, which is to get a collection with a specific route as first (primary) in the collection.  

This makes me question the value of offering this function in the SDK. All we're doing is reordering a collection.  As a developer having to deal with checking the expected result and proceeding accordingly seems overly complicated for a simple reordering of a collection.  We already document that when calling setRoutes() the first route in the collection is considered the primary route. The findRoute() has a lot of value in locating a route based on a map touch. I propose we remove this extension function altogether. To me it offers little value when compared to the complexity involved in using it and there's nothing magical happening in it that requires any navigation SDK specific knowledge.

Reply:
Yea, I agree. Just from this discussion, we can see that one dev would expect a certain effect while another would expect something different, and the docs mention both :joy: Let's remove this and let developers figure out how to re-arrange elements in a collection, the SDK shouldn't have to be concerned about that.